omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PersonResource1_8.java

```diff
@@ -552,4 +556,19 @@ public SimpleObject getAuditInfo(Person person) throws Exception {
 		ret.put("dateCreated", ConversionUtil.convertToRepresentation(person.getPersonDateCreated(), Representation.DEFAULT));
 		return ret;
 	}
+	
+	/**
+	 * Adds the link to patient resource in this person description if person is patient
+	 * 
+	 * @param description whose link set is to be modified
+	 * @return audit resource description
+	 */
+	private DelegatingResourceDescription ifPatientAddLink(DelegatingResourceDescription description) {
+		if (this.person != null && person.isPatient()) {
+			Patient patient = Context.getPatientService().getPatient(this.person.getId());
```

Comment:
What happens if the returned Patient is null?

Reply:
This is a redundant check though I included it. I just felt it safe to include it even when it is addresed by [this line ](https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/DelegatingCrudResource.java#L52). I am now concluding to remove it (this.person != null). Thank you @samuelmale.

Reply:
I think you did get me. I'm not talking about the if statement check but rather the returned `patient`. What happens if the returned value is null?

Reply:
Ohh, good catch @samuelmale. I just didn't expect the related to patient to be null though. In the case this happens, a null pointer exception would be thrown from [this line](https://github.com/openmrs/openmrs-module-webservices.rest/pull/379/commits/f417178211fff0ab8109afe2a7300791a40b15a1#diff-060b1f85f075bfef61fde7654399e5c5R569)

Reply:
Do you wan't to prevent the possibility of a NPE?

Reply:
The exception is adequate enough to describe the problem in the case the unexpected happens. @samuelmale , what other other alternative  would work best?